### ISSUE TITLE
implement cluster state replication

### ISSUE THREAD

**Issue by dmgcodevil** [parapet-io/parapet#49]
We don't need to implement a consensus algorithm (failure-detector base) or an atomic broadcast to reliably replicate a cluster state. For a  leader process whenever it receives a cluster operation (join, leave ) it should send a cluster update to each process and wait for n/2-1 responses. Since a cluster is already using leader election the below algorithm should work:

```
var num_seq = 0
var leader_addr
var state = empty

case req: 
   process(req)
   num_seq  = num_seq  + 1
   UpdateSate(leader_addr, num_seq, state) ~> cluster_processes
   whait for n/2-1 responses

case UpdateSate(addr, n, s):
   if (addr != leader_addr) reject ~> addr
   else if (n < num_seq ) reject ~> addr
   else {
      state = s
      num_seq = n
      ok  ~> addr
   }

case LeaderUpdate(addr):
   leader_addr = addr
   1a GetState ~> cluster_processes
   1b for any state.num_seq  > num_seq  update current state
   1c broacast current state
```

**Comment by dmgcodevil** [comment]
we need coordinator election(CE) algo (can be number based) can be extracted from current leader election. CE can be used in consensus algo we need to run every time cluster elects a new leader.

**Comment by dmgcodevil** [comment]
Implemented in [0.0.1-RC6](https://mvnrepository.com/artifact/io.parapet/core_2.13/0.0.1-RC6)
